src/Polysemy/State/Law.hs

```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE FlexibleInstances     #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+
+module Polysemy.State.Law where
+
+import Polysemy
+import Polysemy.Law
+import Polysemy.State
+import Control.Applicative
+import Control.Arrow
+
+
+------------------------------------------------------------------------------
+-- | A collection of laws that show a `State` interpreter is correct.
+prop_lawfulState
+    :: forall r s
+     . (Eq s, Show s, Arbitrary s, MakeLaw (State s) r)
+    => InterpreterFor (State s) r
+    -> Property
+prop_lawfulState i12n = conjoin
```

Comment:
Nitpick - not sure I understand name `i12n` :smile:

Reply:
**i** nterpretation **n**, but yeah, fair enough